GitTfsTest/Integration/CloneTests.cs

```diff
@@ -60,6 +60,20 @@ public void CloneProjectWithChangesets()
         }
 
         [FactExceptOnUnix]
+        public void CloneWithRecreatedBranches()
+        {
+            h.SetupFake(r =>
+            {
+                r.Changeset(1, "Create Team Project", DateTime.Now).Change(TfsChangeType.Add, TfsItemType.Folder, "$/MyProject");
+                r.Changeset(2, "Create trunk", DateTime.Now).Change(TfsChangeType.Add, TfsItemType.Folder, "$/MyProject/Trunk");
+                r.BranchChangeset(3, "Create branch", DateTime.Now, "$/MyProject/Trunk", "$/MyProject/Branch", 2);
+                r.Changeset(4, "Delete branch", DateTime.Now).Change(TfsChangeType.Delete, TfsItemType.Folder, "$/MyProject/Branch");
+                r.BranchChangeset(5, "Create branch", DateTime.Now, "$/MyProject/Trunk", "$/MyProject/Branch", 4);
```

Comment:
Is this the problem, that the branch got deleted and re-added, but git-tfs didn't notice the deletion, so it didn't remove the branch entry from its list of branches?